### DISCUSSION TITLE
How to access data?

### DISCUSSION THREAD

**Discussion by ederpardeiro** [amidaware/tacticalrmm#1954] [General]
Hi Guys! 

I'm integrating RMM with Grafana to see the health of my servers with a personalized dashboard. I'm finding some problems to filter data, because I found a ready-made template where some fields appearing, but not all, or at least not the fields I need.
In one of the templates, I can filter Clients, Sites and Device, but I can't filter by Server or Workstation and this is a problem for me. I don't need workstations here, I want to filter only servers.
Could I have access in Database? I installed PGAdmin and tryed to access database to explore tables, views and fields, but I was not has success. How can I do it?

Sorry, but I don't speak fluently English.

**Comment by ederpardeiro** [comment]
I'm trying to access using the default port (5432), this is the same port that grafana uses to connect, but using "localhost"
When I tried to access from external network, it didn't work. See configuration in PGAdmin:

![image](https://github.com/user-attachments/assets/1f19fcbd-d586-48a8-8b66-2711f44f2314)

Testing external access, i can see error to access 5432 port:
![image](https://github.com/user-attachments/assets/a70b40a0-2b1f-4772-8ca0-f0ddd983b9c7)

My server has a public IP, so problem isn't redirect.

**Comment by jmoran-shafer** [comment]
I believe this is considered under unsupported, but you should read the
docs (which I believe has grafana instructions) or ask in the Discord under
the Unsupported channel.

**Reply by ederpardeiro** [reply]
Ok, I understand. The problem is filtering this field in Grafana. But for that, I need to access the database. My real question is that: presenting the data in Grafana is just the purpose.

Leaving Grafana aside, is it possible to access the database?

I have the access data (IP, port, database, username, and password), but I can't connect.

Is there any firewall enabled during the installation of Tactical RMM that prevents connection through port 5432? Because my server has a public IP, meaning all ports are open, so I should be able to connect to this port normally.

**Reply by wh1te909** [reply]
This is what our Reporting feature is for, accessing the database and presenting the info in reports, csv etc. There's no firewall enabled during install, that's up to you, but you should not expose port 5432 without locking it down to IP or something. Postgres by default binds only to localhost which is why you can't connect. You'll need to change your postgres config to bind to 0.0.0.0. But this is all unsupported and we can't provide any assistance if you go down this route if something breaks.
